scripts/pre_commit.sh

```diff
@@ -1,15 +1,16 @@
 #!/bin/bash
 set -e
 
-./scripts/build_proto.sh
+# start tests in parallel (with additional parallelism)
+pytest -m fast -n auto &
+
+# check bandit in parallel
+bandit -r src -ll &
+
+# run API documentation test notebooks in parallel
+./scripts/nb_test.sh && pytest examples/api --cov-fail-under 0 &
 
 # fix isort and format with black
-isort .
-black src tests
-pre-commit run --all-files
-bandit -r src -ll
-pytest -m fast
-
-# run API documentation test notebooks
-./scripts/nb_test.sh
-pytest examples/api --cov-fail-under 0
\ No newline at end of file
+./scripts/build_proto.sh && isort . && black src tests && pre-commit run --all-files &
```

Comment:
Hmm...Only the last command will run in the background. You run ```build_proto.sh``` and if successful, you run ```isort```, if success you run...and at the end, you run ```pre-commit run --all-files``` in background.

You use "set -e", this means that you can remove the ```&&```
and simply do:
```
./scripts/build_proto.sh
isort .
black src tests
pre-commit run --all-files  <--- this does not need to run in the background since is the last command
```
if any of the above commands exists with a non zero status code, the script will exit (since you use ```set -e```)

What was the intention here?

Reply:
I guess its just to make it clear they are a single unit, the pre-commit run is the slowest but because its at the end there isnt any advantage, however the slower tests above are run first and backgrounded so that this can at least start as soon as the proto, isort and black are done.

Later if we add another step and it goes after this the pre-commit run --all-files won't block it.

My two goals are:
- as fast as possible
- as clear as possible

Im very interested in any optimal way to re-order things to make it faster. I tried a few different combinations but this was the fastest I came up with.